lib/nmatrix/enumerate.rb

```diff
@@ -73,6 +73,8 @@ def map(&bl)
     return enum_for(:map) unless block_given?
     # NMatrix-jruby currently supports only doubles
     cp  = jruby? ? self : self.cast(dtype: :object)
+    cp.map!(&bl)
```

Comment:
I had previously line 76 and 77 and forgot to add them again. As a result  12 tests were failing for MRI. When I added it again, all tests for MRI were passing.

I have only changed line 75 for JRuby to work.